server/src/main/java/org/elasticsearch/common/io/stream/DelayableWriteable.java

```diff
@@ -28,10 +28,23 @@
 import java.util.function.Supplier;
 
 /**
- * A holder for {@link Writeable}s that can delays reading the underlying
- * {@linkplain Writeable} when it is read from a remote node.
+ * A holder for {@link Writeable}s that delays reading the underlying object
+ * on the receiving end. To be used for objects that use a lot of memory hence
+ * it is desirable to keep them around only for a limited amount of time.
```

Comment:
Its more that deserializing it is inefficient compared to leaving it serialized. "Big" objects are fine to deserialize if they are no bigger when you deserialize them.

Reply:
good point, will rephrase.